src/main/java/org/folio/linked/data/service/resource/ResourceMarcServiceImpl.java

```diff
@@ -98,17 +100,25 @@ public Boolean isSupportedByInventoryId(String inventoryId) {
 
   @Override
   public ResourceResponseDto getResourcePreviewByInventoryId(String inventoryId) {
-    var response = srsClient.getFormattedSourceStorageInstanceRecordById(inventoryId);
-    return Optional.ofNullable(response.getBody())
-      .map(Record::getParsedRecord)
-      .map(ParsedRecord::getContent)
-      .map(this::toJsonString)
-      .flatMap(marcBib2ldMapper::fromMarcJson)
+    return getResource(inventoryId)
       .map(resourceModelMapper::toEntity)
       .map(resourceDtoMapper::toDto)
       .orElse(null);
   }
 
+  @Override
+  public ResourceIdDto importMarcRecord(String inventoryId) {
+    return getResource(inventoryId)
+      .map(resource -> {
+        resource.getFolioMetadata().setSource(LINKED_DATA);
+        return resource;
+      })
+      .map(this::saveMarcResource)
+      .map(String::valueOf)
+      .map(id -> new ResourceIdDto().id(id))
+      .orElse(null);
```

Comment:
i'd suggest to return empty Optional here instead of null. ResponseEntity should work fine with Optional objects

Reply:
done

Reply:
ResponseEntity<Optional<ResourceIdDto>> can automatically solve this issue as per the documentation:
https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/ResponseEntity.html#of(java.util.Optional)

it will lead to 404 NOT_FOUND in case of Optional.empty(), so we don't need to throw an exception unless we want to customize handling

Reply:
@AndreiBordak, I don't see a way to specify Optional using swagger